### DISCUSSION TITLE
Vue middleware - Constraining by HTML Element boundary

### DISCUSSION THREAD

**Discussion by Ericlm** [floating-ui/floating-ui#2564] [Q&A] [answered]
Hello there! I'm currently working with Floating UI for Vue, and I'm struggling to create a custom middleware for constraining a floating element to be contained within another HTMLElement. 
After reading the docs, I don't really understand how custom middlewares should be declared (especially what their types should be). 
Could you please help me? 😀

**Comment by atomiks** [comment, marked as answer]
Are you sure you need a custom one? `flip` & `shift` both accept a `boundary` prop to do that. The `Middleware` type can be used to create a custom one though: `const middleware: Middleware = {};`

**Comment by Ericlm** [comment]
Thank you very much, it works perfectly! Didn't even know those middlewares had those props 😃
